packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -927,9 +927,11 @@ class _PlatformViewLinkState extends State<PlatformViewLink> {
   }
 
   void _onPlatformViewCreated(int id) {
-    setState(() {
-      _platformViewCreated = true;
-    });
+    if (mounted) {
```

Comment:
This might be a dumb question but when can this be called and not mounted? What side effects are there if setState is not called because this view was not mounted?

Reply:
> This might be a dumb question but when can this be called and not mounted?

While working on the fix, the call to `setState` was the place where I started, since that was causing the exceptions. Guarding the cal with `if (mounted)` resolves the issue, but I asked myself the same question. The root cause is that native platform views are created asynchronously through a `MethodChannel` but the `PlatformViewLink` that triggered the creation of the platform view can be disposed before the platform view has been created. A `PlatformViewLink` disposes its `PlatformViewController` and it would make sense for that to prevent any further callbacks from firing. If that is the case, the call to `setState` should not require a guard. Looking at `AndroidViewController.dispose` I found that that is not guaranteed, so I changed its implementation to synchronously clean up all listeners.

Fixing `AndroidViewController` makes the guard for `setState` unnecessary, but `AndroidViewController` is not the only implementation of `PlatformViewController` so to be defensive I thought it might make sense to leave it in.